operator/builtin/input/azure/eventhub/event_hub.go

```diff
@@ -162,26 +162,24 @@ func (e *EventHubInput) startConsumer(ctx context.Context, partitionID string, h
 	}
 
 	offsetStr := ""
-	if e.startAtEnd {
-		offset, err := e.persist.Read(e.namespace, e.name, e.group, partitionID)
-		if err != nil {
-			x := fmt.Sprintf("Error while reading offset for partition_id %s, starting at end", partitionID)
-			e.Errorw(x, zap.Error(err))
-		} else {
-			offsetStr = offset.Offset
-		}
+	offset, err := e.persist.Read(e.namespace, e.name, e.group, partitionID)
+	if err != nil {
+		x := fmt.Sprintf("Error while reading offset for partition_id %s, starting at end", partitionID)
+		e.Errorw(x, zap.Error(err))
```

Comment:
Just to confirm - in the event of an error, the intention here is to continue on?

Reply:
I think so. I believe the only time an error will be returned is if we fail to unmarshal the json object being read from the database. In the case of corruption, should we log the error and "start at end"?

persistence.go
```
func (p *Persister) Read(namespace, name, consumerGroup, partitionID string) (persist.Checkpoint, error) {
...
	var checkpoint persist.Checkpoint
	err := json.Unmarshal(value, &checkpoint)
	return checkpoint, err
```